### ISSUE TITLE
Validation for +49 7775abc2219

### ISSUE THREAD

**Issue by pquerner** [googlei18n/libphonenumber#1397]
+49 7775abc2219 seems to be "valid" when its not.

**Comment by kkeshava** [comment]
@pquerner,
Thank you for using libphonenumber!

We need more information to understand the issue. Did you mean to say that phone numbers cannot have alpha characters? Or the numbers of the form 7775xxx2219 are invalid? 

Please let us know as to why you think 7775xxx2219 should be invalid? Please see the [contribution guidelines](https://github.com/googlei18n/libphonenumber/blob/master/CONTRIBUTING.md).

The phone numbers need not be only digits. In cases like 1800 MICROSOFT, the characters are converted to digits. 
Note - Please also see [Falsehoods Programmers Believe about Phone Numbers](https://github.com/googlei18n/libphonenumber/blob/master/FALSEHOODS.md) for more information.

**Comment by pquerner** [comment]
Thanks for the questions  @kkeshava 

I know its common in some countries to use characters instead of digits, but it isnt very common (I've never seen it) in Germany, therefore the validation process should keep that in mind I think.

In the end we wrote regex validation our selfs.

Problem is:
- The input is used for order/shipping information. 
- Behind the shop system is another system which cannot handle this type of phone number (one issue we solved backend-wise of course)
- In the above step we couldnt fully rely on libphonenumber as is, therefore this issue was created.


TL;DR
Validation _could_ be improved by having some sort of Country rules present.
Ie. its uncommon in Germany to have characters in a telephone number (see Ad in a newspaper or something)

**Comment by keghani** [comment]
Thanks for the questions @pquerner!

It sounds like you're keeping the input string entered by the user and sending that to the other system if we consider it valid.
I think this is solved by storing the number as a PhoneNumber in your system; and whenever you need it as a string call a formatting method from PhoneNumberUtil.

Please see the documentation in [PhoneNumberUtil.java](https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java). There are supported ways to customize the behavior you want for your code, e.g. see "isAlphaNumber" which you might be able to use to filter out these numbers. But for purposes of validation we have defined isValidNumber to allow alpha numbers.

I'll close this as it's not an issue with the library. Please consider joining the [discussion group](https://groups.google.com/forum/#!forum/libphonenumber-discuss) if you'd like to discuss this further.
